Extensions/TopDownMovementBehavior/tests/topdownmovementbehavior.spec.js

```diff
@@ -0,0 +1,372 @@
+// @ts-check
+describe('gdjs.TopDownMovementRuntimeBehavior', function () {
+  const topDownName = 'auto1';
+
+  const createScene = () => {
+    const runtimeGame = new gdjs.RuntimeGame({
+      variables: [],
+      // @ts-ignore - missing properties.
+      properties: { windowWidth: 800, windowHeight: 600 },
+      resources: { resources: [] },
+    });
+    const runtimeScene = new gdjs.RuntimeScene(runtimeGame);
+    runtimeScene.loadFromScene({
+      layers: [
+        {
+          name: '',
+          visibility: true,
+          effects: [],
+          cameras: [],
+
+          ambientLightColorR: 0,
+          ambientLightColorG: 0,
+          ambientLightColorB: 0,
+          isLightingLayer: false,
+          followBaseLayerCamera: true,
+        },
+      ],
+      variables: [],
+      r: 0,
+      v: 0,
+      b: 0,
+      mangledName: 'Scene1',
+      name: 'Scene1',
+      stopSoundsOnStartup: false,
+      title: '',
+      behaviorsSharedData: [],
+      objects: [],
+      instances: [],
+    });
+    runtimeScene._timeManager.getElapsedTime = function () {
+      return (1 / 60) * 1000;
+    };
+    return runtimeScene;
+  };
+
+  const addPlayer = (runtimeScene, allowDiagonals) => {
+    const player = new gdjs.RuntimeObject(runtimeScene, {
```

Comment:
Nitpicking: I now recommend to use `gdjs.TestRuntimeObject`, which has a function `setCustomWidthAndHeight`. Better than overriding getWidth/getHeight :) 
Example here: https://github.com/4ian/GDevelop/blob/479167638b647527996f1b782cdeffeef5381cf7/GDJS/tests/tests/runtimeobject.js#L39-L44